### ISSUE TITLE
"transparent mode" like mitmproxy

### ISSUE THREAD

**Issue by hey-jude** [james-proxy/james#328]
I need "transparent mode" like mitmproxy.
http://docs.mitmproxy.org/en/stable/transparent.html

It's very useful, especially device cannot be assigned with proxy address.

**Comment by mitchhentges** [comment]
Hmm, interesting.
@tomitm and I are doing a bit of research, and [filternet](https://github.com/axiak/filternet) might be useful.
Currently, I'm a little knee-deep in university, but PR's welcome, of course. Maybe Tomi will even bang it out in his spare time :smile:

**Comment by tomitm** [comment]
Yeah, `filternet` seems like an alternative to `hoxy`, which we currently use. We should land support in `hoxy` (since really, James is mostly a UI for it) - it doesn't look like `filternet` has been updated since 2014.

**Comment by hey-jude** [comment]
Mitmproxy is implemented that just redirects requests on ethernet to local loopback on transparent mode. Platform's firewall takes on packet fltering-redirecting. It's iptables on linux, pf on macos.

That way seems not clean, but works well.
